NETCORE/src/Shared/Implementation/TelemetryConfigurationOptionsSetup.cs

```diff
@@ -255,7 +255,12 @@ public void Configure(TelemetryConfiguration configuration)
                     }
                     catch (Exception ex)
                     {
-                        throw new Exception($"Failed to initialize module '{module.GetType()}'", ex);
+                        var tempEx = new Exception($"Failed to initialize module '{module.GetType()}'", ex);
+#if AI_ASPNETCORE_WEB
+                        AspNetCoreEventSource.Instance.TelemetryConfigurationSetupFailure(tempEx.ToInvariantString());
```

Comment:
let add a new method `TelemetryModuleInitialziationSetupFailure` with name of module as an argument. That way we can easily isolate issues if one module goes rogue.

Reply:
An error has occurred while initializing {modulename}. Functionalities offered by this module will be affected. -- something like this